### ISSUE TITLE
How to parameterize a rule/OVAL without breaking existing use?

### ISSUE THREAD

**Issue by Caligatio** [ComplianceAsCode/content#6408]
I'm working on implementing the CIS Level 1 profile for Ubuntu 20.04 and its guidance has changed on the SSH `ClientAliveCountMax` value. It appears in all the existing rules that it is recommended to be set to `0` whereas in this guidance it is set to `3`.  I had previously implemented the Ubuntu 20.04 "Standard" profile which used this set to `0` recommendation.

How to I change the rule to allow passing in a `xccdf_value` but also having default value so I do not need to change all the pre-existing rules?  Since I already used an instance of it being `0`, I rather not set this at the `prodtype` level.

**Comment by ggbecker** [comment]
This rule has been recently changed to have `0` hardcoded, because any other value doesn't make sense if the purpose is to logout inactive users when the timeout occurs. It was quite controversial and probably CIS will have to update its benchmarks.

I really recommend reading this: https://bugzilla.redhat.com/show_bug.cgi?id=1873547 and the pull request that made the change: https://github.com/ComplianceAsCode/content/pull/6293

As a last resource you could revert some of the changes and use the variable: https://github.com/ComplianceAsCode/content/blob/master/linux_os/guide/services/ssh/var_sshd_set_keepalive.var#L0-L1

to refine the value on a profile level by adding:

`- var_sshd_set_keepalive=3` to the `selections` list

to the `cis.profile` you are building

**Comment by Caligatio** [comment]
Hmm, that thread doesn't agree with the docs:

```
ClientAliveCountMax
             Sets the number of client alive messages which may be sent with‐
             out sshd(8) receiving any messages back from the client.  If this
             threshold is reached while client alive messages are being sent,
             sshd will disconnect the client, terminating the session.  It is
             important to note that the use of client alive messages is very
             different from TCPKeepAlive.  The client alive messages are sent
             through the encrypted channel and therefore will not be spoofa‐
             ble.  The TCP keepalive option enabled by TCPKeepAlive is spoofa‐
             ble.  The client alive mechanism is valuable when the client or
             server depend on knowing when a connection has become unrespon‐
             sive.

             The default value is 3.  If ClientAliveInterval is set to 15, and
             ClientAliveCountMax is left at the default, unresponsive SSH
             clients will be disconnected after approximately 45 seconds.
             Setting a zero ClientAliveCountMax disables connection termina‐
             tion.
```

The critical part is the last sentence of the second paragraph: **Setting a zero ClientAliveCountMax disables connection termination**.

Any guidance here on how to proceed?  The decision to hardcode this to a `0` seems to explicitly disagree with the stated intent (if I'm read this all correctly).

**Comment by ggbecker** [comment]
The problem with this SSH configuration is that it doesn't disconnects idle user, it disconnects unresponsive connections which may be caused by network connection issues. The original intent for this rule was to disconnect idle users and the only way that it could be achieved was `abusing` this configuration by setting to `0`. When we hardcoded `0`, we had mainly RHEL8 in mind, which still has openssh 8.0 thus containing this behaviour.

IMO, this security requirement should be revisited and maybe even changed to something else. Multiple policies might be affected.

Your suggested change: https://github.com/ComplianceAsCode/content/pull/6416/files#diff-2f6a21382133b2d4c0760126d427b455dd0b61e5088068813d8114b975b5550fR68

is not the best approach since it will spread the same value across all profiles supported by ubuntu2004, at the moment it might not be a big deal, but maybe in the future it can be. The best approach would be to revert back the changes we made and make the rule to use the variable again as I mentioned in [comment](https://github.com/ComplianceAsCode/content/issues/6408#issuecomment-732254746) above.

Big thank you for pointing out this SSH behaviour change.

I hope it clarifies a bit more the issue.
